linker/Tests/TestCasesRunner/TestCaseMetadaProvider.cs

```diff
@@ -49,13 +50,21 @@ public virtual TestCaseLinkerOptions GetLinkerOptions ()
 			return tclo;
 		}
 
-		public virtual IEnumerable<string> GetReferencedAssemblies (NPath workingDirectory)
+		public virtual IEnumerable<string> GetCommonReferencedAssemblies (NPath workingDirectory)
 		{
 			yield return workingDirectory.Combine ("Mono.Linker.Tests.Cases.Expectations.dll").ToString ();
 			yield return "mscorlib.dll";
+		}
 
+		public virtual IEnumerable<string> GetReferencedAssemblies (NPath workingDirectory)
+		{
 			foreach (var referenceAttr in _testCaseTypeDefinition.CustomAttributes.Where (attr => attr.AttributeType.Name == nameof (ReferenceAttribute))) {
-				yield return (string) referenceAttr.ConstructorArguments.First ().Value;
+				var fileName = (string) referenceAttr.ConstructorArguments.First ().Value;
+				if (fileName.StartsWith("System.") || fileName.StartsWith("Mono.") || fileName.StartsWith("Microsoft."))
```

Comment:
Nit: missing space before (
Would be nice to use OrdinarComparer

Reply:
Fixed